app/Rest/Employees_Controller.php

```diff
@@ -0,0 +1,475 @@
+			'description' => __( 'Ensure result set excludes specific IDs.' ),
+			'type'        => 'array',
+			'items'       => array(
+				'type' => 'integer',
+			),
+			'default'     => array(),
+		);
+
+		$query_params['include'] = array(
+			'description' => __( 'Limit result set to specific IDs.' ),
+			'type'        => 'array',
+			'items'       => array(
+				'type' => 'integer',
+			),
+			'default'     => array(),
+		);
+
+		$query_params['offset'] = array(
+			'description' => __( 'Offset the result set by a specific number of items.' ),
+			'type'        => 'integer',
+		);
+
+		$query_params['order'] = array(
+			'description' => __( 'Order sort attribute ascending or descending.' ),
+			'type'        => 'string',
+			'default'     => 'desc',
+			'enum'        => array( 'asc', 'desc' ),
+		);
+
+		$query_params['orderby'] = array(
+			'description' => __( 'Sort collection by object attribute.' ),
+			'type'        => 'string',
+			'default'     => 'date',
+			'enum'        => array(
+				'author',
+				'date',
+				'id',
+				'include',
+				'modified',
+				'parent',
+				'relevance',
+				'slug',
+				'include_slugs',
+				'title',
+			),
+		);
+
+		return $query_params;
+	}
+
+	/**
+	 * Get_item_schema
+	 *
+	 * @return array
+	 */
+	public function get_item_schema() {
+		$schema = array(
+			'$schema'    => 'http://json-schema.org/draft-04/schema#',
+			'title'      => Employee_Post_Type::$ID,
+			'type'       => 'object',
+			'properties' => array(
+				'title'     => array(
+					'description' => sprintf( __( 'The title for the %s.' ), Employee_Post_Type::$ID ),
+					'type'        => 'string',
+				),
+				'content'   => array(
+					'description' => sprintf( __( 'The content for the %s.' ), Employee_Post_Type::$ID ),
+					'type'        => 'string',
+				),
+				'acf_field' => array(
+					'description' => sprintf( __( 'The ACF field for the %s.' ), Employee_Post_Type::$ID ),
+					'type'        => 'object',
+				),
+			),
+		);
+
+		return $this->add_additional_fields_schema( $schema );
+	}
+
+	/**
+	 * Prepare_item_for_database
+	 *
+	 * @param \WP_REST_Request $request Full details about the request.
+	 *
+	 * @return object|\stdClass|\WP_Error
+	 */
+	protected function prepare_item_for_database( $request ) {
+		$prepared_post = new \stdClass();
+
+		if ( isset( $request['id'] ) ) {
+			$existing_post = $this->get_employee( $request['id'] );
+			if ( is_wp_error( $existing_post ) ) {
+				return $existing_post;
+			}
+
+			$prepared_post->ID = $existing_post->ID;
+		}
+
+		$schema = $this->get_item_schema();
+
+		if ( ! empty( $schema['properties']['title'] ) && isset( $request['title'] ) ) {
+			if ( is_string( $request['title'] ) ) {
+				$prepared_post->post_title = $request['title'];
+			}
+		}
+
+		if ( ! empty( $schema['properties']['content'] ) && isset( $request['content'] ) ) {
+			if ( is_string( $request['content'] ) ) {
+				$prepared_post->post_content = $request['content'];
+			}
+		}
```

Comment:
writing a lot of "if" is not very good.

you can write foreach($schema['properties'] as $prop) {}
and then check for request.

then you can validate input for required fields (in schema you specify types and required flag), so in case there is an error - you can return WP_Error that data is not valid.

such method can be placed in generic REST_Controller like $this->parse_request($request, $schema)

Reply:
not sure why we're not checking post meta here.